core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts

```diff
@@ -101,6 +101,41 @@ angular.module('oppia').component('contributorDashboardPage', {
         ctrl.userCanReviewTranslationSuggestionsInLanguages = [];
         ctrl.userCanReviewVoiceoverSuggestionsInLanguages = [];
         ctrl.userCanReviewQuestions = false;
+        ctrl.defaultHeaderVisible = true;
+        ctrl.collapsibleHeaderVisible = false;
+        ctrl.scrollTest = false;
+
+        var $win = angular.element($window);
+        var defaultHeader = angular.element(document.querySelector(
+          '#default-header'));
+        var collapsibleHeader = angular.element(document.querySelector(
+          '#collapsible-header'));
+
+        window.onscroll = function() {
+          ctrl.scrollFunction();
+        };
```

Comment:
Instead of modifying `window` can we add an event listener to the injected `$window`? e.g.

```
$window.addEventListener('scroll', function() {
  ...
});
```

See core/templates/pages/exploration-player-page/learner-experience/conversation-skin.directive.ts for an example.

Reply:
Used your suggestion and done.